In [1]:
import pandas as pd

**Dispaly data**

In [47]:
df = pd.read_csv('/content/Ecommerce Purchases')


**1. Display top 5 rows of the dataset**

In [49]:
df.head(5)

Address    Lot AM or PM  \
0  16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1  9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
2                   Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   
3              7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   
4  23012 Munoz Drive Suite 337\nNew Cynthia, TX 5...  20 IE       AM   

                                        Browser Info  \
0  Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1  Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
2  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
3  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   
4  Opera/9.58.(X11; Linux x86_64; it-IT) Presto/2...   

                           Company       Credit Card CC Exp Date  \
0                  Martinez-Herman  6011929061123406       02/20   
1  Fletcher, Richards and Whitaker  3337758169645356       11/18   
2       Simpson, Williams and Pham      675957666125       08/19   
3  Williams, Marshall and Buchanan  6011578504430710       02/24   
4        Brown, Watson and Andrews  6011456623207998       10/25   

   CC Security Code                  CC Provider  \
0               900                 JCB 16 digit   
1               561                   Mastercard   
2               699                 JCB 16 digit   
3               384                     Discover   
4               678  Diners Club / Carte Blanche   

                            Email                                     Job  \
0               pdunlap@yahoo.com  Scientist, product/process development   
1              anthony41@reed.com                       Drilling engineer   
2  amymiller@morales-harrison.com                Customer service manager   
3     brent16@olson-robinson.info                       Drilling engineer   
4     christopherwright@gmail.com                             Fine artist   

        IP Address Language  Purchase Price  
0  149.146.147.205       el           98.14  
1     15.160.41.51       fr           70.73  
2   132.207.160.22       de            0.95  
3     30.250.74.19       es           78.04  
4     24.140.33.94       es           77.82

**2. check last 5 rows of dataset**

In [48]:
df.tail(5)

Address    Lot AM or PM  \
9995      966 Castaneda Locks\nWest Juliafurt, CO 96415  92 XI       PM   
9996  832 Curtis Dam Suite 785\nNorth Edwardburgh, T...  41 JY       AM   
9997              Unit 4434 Box 6343\nDPO AE 28026-0283  74 Zh       AM   
9998               0096 English Rest\nRoystad, IA 12457  74 cL       PM   
9999     40674 Barrett Stravenue\nGrimesville, WI 79682  64 Hr       AM   

                                           Browser Info  \
9995  Mozilla/5.0 (Windows NT 5.1) AppleWebKit/5352 ...   
9996  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
9997  Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_7...   
9998  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_8;...   
9999  Mozilla/5.0 (X11; Linux i686; rv:1.9.5.20) Gec...   

                       Company       Credit Card CC Exp Date  \
9995             Randall-Sloan   342945015358701       03/22   
9996  Hale, Collins and Wilson   210033169205009       07/25   
9997              Anderson Ltd  6011539787356311       05/21   
9998                  Cook Inc   180003348082930       11/17   
9999                Greene Inc  4139972901927273       02/19   

      CC Security Code       CC Provider                    Email  \
9995               838      JCB 15 digit   iscott@wade-garner.com   
9996               207      JCB 16 digit       mary85@hotmail.com   
9997                 1     VISA 16 digit        tyler16@gmail.com   
9998               987  American Express  elizabethmoore@reid.net   
9999               302      JCB 15 digit    rachelford@vaughn.com   

                           Job       IP Address Language  Purchase Price  
9995                Printmaker    29.73.197.114       it           82.21  
9996           Energy engineer   121.133.168.51       pt           25.63  
9997        Veterinary surgeon    156.210.0.254       el           83.98  
9998  Local government officer     55.78.26.143       es           38.84  
9999    Embryologist, clinical  176.119.198.199       el           67.59

**3. Check Data Type of each column**

In [5]:
df.dtypes

Address              object
Lot                  object
AM or PM             object
Browser Info         object
Company              object
Credit Card           int64
CC Exp Date          object
CC Security Code      int64
CC Provider          object
Email                object
Job                  object
IP Address           object
Language             object
Purchase Price      float64
dtype: object

**4. check null values in the datasets**

In [6]:
df.isnull().sum() # to check if there is any missing rows in the datasets/.

Address             0
Lot                 0
AM or PM            0
Browser Info        0
Company             0
Credit Card         0
CC Exp Date         0
CC Security Code    0
CC Provider         0
Email               0
Job                 0
IP Address          0
Language            0
Purchase Price      0
dtype: int64

**5. How many rows and columns in the datasets**

In [7]:
len(df.columns)

14

In [8]:
len(df) # total data

10000

In [9]:
df.info() # gives you all the information of the  data. 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Address           10000 non-null  object 
 1   Lot               10000 non-null  object 
 2   AM or PM          10000 non-null  object 
 3   Browser Info      10000 non-null  object 
 4   Company           10000 non-null  object 
 5   Credit Card       10000 non-null  int64  
 6   CC Exp Date       10000 non-null  object 
 7   CC Security Code  10000 non-null  int64  
 8   CC Provider       10000 non-null  object 
 9   Email             10000 non-null  object 
 10  Job               10000 non-null  object 
 11  IP Address        10000 non-null  object 
 12  Language          10000 non-null  object 
 13  Purchase Price    10000 non-null  float64
dtypes: float64(1), int64(2), object(11)
memory usage: 1.1+ MB


**6. Higest and lowest purcahse price**

In [10]:
df['Purchase Price'].max()

99.99

In [11]:
df['Purchase Price'].min()

0.0

**7. Average purchase price**

In [12]:
df['Purchase Price'].mean()

50.34730200000025

**8. How many people have fr as their language** 

In [13]:
df.columns

Index(['Address', 'Lot', 'AM or PM', 'Browser Info', 'Company', 'Credit Card',
       'CC Exp Date', 'CC Security Code', 'CC Provider', 'Email', 'Job',
       'IP Address', 'Language', 'Purchase Price'],
      dtype='object')

In [14]:
len(df[df['Language'] == 'fr'])

1097

In [15]:
df[df['Language'] == 'fr'].count()

Address             1097
Lot                 1097
AM or PM            1097
Browser Info        1097
Company             1097
Credit Card         1097
CC Exp Date         1097
CC Security Code    1097
CC Provider         1097
Email               1097
Job                 1097
IP Address          1097
Language            1097
Purchase Price      1097
dtype: int64

**9. Job title contain eng**

In [16]:
df.columns

Index(['Address', 'Lot', 'AM or PM', 'Browser Info', 'Company', 'Credit Card',
       'CC Exp Date', 'CC Security Code', 'CC Provider', 'Email', 'Job',
       'IP Address', 'Language', 'Purchase Price'],
      dtype='object')

In [17]:
len(df[df['Job'].str.contains('engineer', case = False)])

984

**10. Find the email of the person with the followiin ip address : 132 207 160 22**

In [18]:
df[df['IP Address']== '132.207.160.22']['Email']

2    amymiller@morales-harrison.com
Name: Email, dtype: object

**11. How many people have master as therie credit card provider and made a purchase abve 50**

In [ ]:
df.columns

Index(['Address', 'Lot', 'AM or PM', 'Browser Info', 'Company', 'Credit Card',
       'CC Exp Date', 'CC Security Code', 'CC Provider', 'Email', 'Job',
       'IP Address', 'Language', 'Purchase Price'],
      dtype='object')

In [ ]:
len(df[(df['CC Provider'] == 'Mastercard') & (df['Purchase Price']> 50)])

405

**12. Find email of the person with the following credit card number: 466482525897302**

In [19]:
df.columns

Index(['Address', 'Lot', 'AM or PM', 'Browser Info', 'Company', 'Credit Card',
       'CC Exp Date', 'CC Security Code', 'CC Provider', 'Email', 'Job',
       'IP Address', 'Language', 'Purchase Price'],
      dtype='object')

In [23]:
df[df['Credit Card'] ==4664825258997302] ['Email'] # the email address you want

9992    bberry@wright.net
Name: Email, dtype: object

**13. How many people purchase during AM and how many people purchase during PM**

In [50]:
print(len(df[df['AM or PM']  == 'AM']))

4932


In [51]:
print(len(df[df['AM or PM']  == 'PM']))

5068


In [28]:
# another way
df['AM or PM'].value_counts()

PM    5068
AM    4932
Name: AM or PM, dtype: int64

**14. How many people have a credit card that expired in 2020**

In [30]:
df['CC Exp Date']  # see the format of the data.

0       02/20
1       11/18
2       08/19
3       02/24
4       10/25
        ...  
9995    03/22
9996    07/25
9997    05/21
9998    11/17
9999    02/19
Name: CC Exp Date, Length: 10000, dtype: object

In [37]:
def fun():
  count = 0
  for date in df['CC Exp Date']:
    if date.split('/')[1]== '20':
      count = count + 1
  print(count)

In [38]:
fun()

988


**15. Top 5 most popular email providers (3g gmail.com, yahoo.com, etc...)**

In [40]:
list1 = []
for email in df['Email']:
  list1.append(email.split("@")[1])

In [41]:
df['temp'] = list1
df.head(2)

Address    Lot AM or PM  \
0  16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1  9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   

                                        Browser Info  \
0  Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1  Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   

                           Company       Credit Card CC Exp Date  \
0                  Martinez-Herman  6011929061123406       02/20   
1  Fletcher, Richards and Whitaker  3337758169645356       11/18   

   CC Security Code   CC Provider               Email  \
0               900  JCB 16 digit   pdunlap@yahoo.com   
1               561    Mastercard  anthony41@reed.com   

                                      Job       IP Address Language  \
0  Scientist, product/process development  149.146.147.205       el   
1                       Drilling engineer     15.160.41.51       fr   

   Purchase Price       temp  
0           98.14  yahoo.com  
1           70.73   reed.com

In [46]:
df['temp'].value_counts().head(7)

hotmail.com     1638
yahoo.com       1616
gmail.com       1605
smith.com         42
williams.com      37
brown.com         29
johnson.com       29
Name: temp, dtype: int64